src/views/Player.vue

```diff
@@ -200,12 +213,18 @@ export default {
 
       // TODO: make this better -> not using the
       // benefits of Vue here
-      var index = Number(ivq.id)
+      var currQuesIndex = Number(ivq.id)
 
       // Checking if the object is empty or not.
       // If empty, push the answer. Otherwise don't.
-      if(Object.keys(this.answers[index]).length === 0)
-        this.answers[index] = answer
+      if(Object.keys(this.answers[currQuesIndex]).length === 0){
+        this.answers[currQuesIndex] = answer
+      }
+
+      this.submitted[currQuesIndex] += 1;
+
+      this.skipped[currQuesIndex] = 0;
+      this.revised[currQuesIndex] = 0;
```

Comment:
what if someone skipped and came back and answered?
I don't think other two arrays need to be affected by the action - just update the list corresponding to the action

Reply:
Oopar likha tha ye

"This is done because if a user goes back to an earlier question, and makes another interaction with the buttons, we might have overlapping information. Like -> for question 1, we will see that the user has done all three, submit, skip, and revise."

Depends how should we record. A user can skip and come back and answer -> current logic se uska skip array mein 0 ho jayega (jo pehle 1 tha) and answered array mein 1 ho jayega

Agar hum baaki dono arrays ko update nhi karenge, then uska skip mein bhi 1 rahega and submit mein bhi 1. Toh kaise karna hai?